File: binding/src/libqfuturewatcher.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qfuturewatcherbase.html
pub const qfuturewatcherbase = struct {

    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QFutureWatcherBase_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qfuturewatcherbase.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn ProgressValue(self: ?*anyopaque, ) i32 {
        return qtc.QFutureWatcherBase_ProgressValue(@ptrCast(self));
    }

    pub fn ProgressMinimum(self: ?*anyopaque, ) i32 {
        return qtc.QFutureWatcherBase_ProgressMinimum(@ptrCast(self));
    }

    pub fn ProgressMaximum(self: ?*anyopaque, ) i32 {
        return qtc.QFutureWatcherBase_ProgressMaximum(@ptrCast(self));
    }

    pub fn ProgressText(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QFutureWatcherBase_ProgressText(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qfuturewatcherbase.ProgressText: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn IsStarted(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsStarted(@ptrCast(self));
    }

    pub fn IsFinished(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsFinished(@ptrCast(self));
    }

    pub fn IsRunning(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsRunning(@ptrCast(self));
    }

    pub fn IsCanceled(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsCanceled(@ptrCast(self));
    }

    pub fn IsPaused(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsPaused(@ptrCast(self));
    }

    pub fn IsSuspending(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsSuspending(@ptrCast(self));
    }

    pub fn IsSuspended(self: ?*anyopaque, ) bool {
        return qtc.QFutureWatcherBase_IsSuspended(@ptrCast(self));
    }

    pub fn WaitForFinished(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_WaitForFinished(@ptrCast(self));
    }

    pub fn SetPendingResultsLimit(self: ?*anyopaque, limit: i32) void {
        qtc.QFutureWatcherBase_SetPendingResultsLimit(@ptrCast(self), limit);
    }

    pub fn Event(self: ?*anyopaque, event: ?*anyopaque) bool {
        return qtc.QFutureWatcherBase_Event(@ptrCast(self), @ptrCast(event));
    }

    pub fn Started(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Started(@ptrCast(self));
    }

    pub fn OnStarted(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Started(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Finished(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Finished(@ptrCast(self));
    }

    pub fn OnFinished(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Finished(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Canceled(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Canceled(@ptrCast(self));
    }

    pub fn OnCanceled(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Canceled(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Paused(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Paused(@ptrCast(self));
    }

    pub fn OnPaused(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Paused(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Suspending(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Suspending(@ptrCast(self));
    }

    pub fn OnSuspending(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Suspending(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Suspended(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Suspended(@ptrCast(self));
    }

    pub fn OnSuspended(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Suspended(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Resumed(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Resumed(@ptrCast(self));
    }

    pub fn OnResumed(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_Resumed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ResultReadyAt(self: ?*anyopaque, resultIndex: i32) void {
        qtc.QFutureWatcherBase_ResultReadyAt(@ptrCast(self), resultIndex);
    }

    pub fn OnResultReadyAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_ResultReadyAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ResultsReadyAt(self: ?*anyopaque, beginIndex: i32, endIndex: i32) void {
        qtc.QFutureWatcherBase_ResultsReadyAt(@ptrCast(self), beginIndex, endIndex);
    }

    pub fn OnResultsReadyAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, i32) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_ResultsReadyAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ProgressRangeChanged(self: ?*anyopaque, minimum: i32, maximum: i32) void {
        qtc.QFutureWatcherBase_ProgressRangeChanged(@ptrCast(self), minimum, maximum);
    }

    pub fn OnProgressRangeChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, i32) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_ProgressRangeChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ProgressValueChanged(self: ?*anyopaque, progressValue: i32) void {
        qtc.QFutureWatcherBase_ProgressValueChanged(@ptrCast(self), progressValue);
    }

    pub fn OnProgressValueChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_ProgressValueChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ProgressTextChanged(self: ?*anyopaque, progressText: []const u8) void {
        const progressText_str = qtc.libqt_string{
    .len = progressText.len,
    .data = progressText.ptr,
};
qtc.QFutureWatcherBase_ProgressTextChanged(@ptrCast(self), progressText_str);
    }

    pub fn OnProgressTextChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.QFutureWatcherBase_Connect_ProgressTextChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Cancel(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Cancel(@ptrCast(self));
    }

    pub fn SetSuspended(self: ?*anyopaque, suspendVal: bool) void {
        qtc.QFutureWatcherBase_SetSuspended(@ptrCast(self), suspendVal);
    }

    pub fn Suspend(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Suspend(@ptrCast(self));
    }

    pub fn Resume(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Resume(@ptrCast(self));
    }

    pub fn ToggleSuspended(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_ToggleSuspended(@ptrCast(self));
    }

    pub fn SetPaused(self: ?*anyopaque, paused: bool) void {
        qtc.QFutureWatcherBase_SetPaused(@ptrCast(self), paused);
    }

    pub fn Pause(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_Pause(@ptrCast(self));
    }

    pub fn TogglePaused(self: ?*anyopaque, ) void {
        qtc.QFutureWatcherBase_TogglePaused(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QFutureWatcherBase_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qfuturewatcherbase.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QFutureWatcherBase_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qfuturewatcherbase.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QFutureWatcherBase_Delete(@ptrCast(self));
    }
};

File: binding/src/libqfuturewatcher.cpp
#include <QEvent>
#include <QFutureWatcherBase>
#include <QMetaMethod>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qfuturewatcher.h>
#include "libqfuturewatcher.h"
#include "libqfuturewatcher.hxx"

libqt_string QFutureWatcherBase_Tr(const char* s) {
	QString _ret = QFutureWatcherBase::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QFutureWatcherBase_ProgressValue(const QFutureWatcherBase* self) {
	return self->progressValue();
}

int QFutureWatcherBase_ProgressMinimum(const QFutureWatcherBase* self) {
	return self->progressMinimum();
}

int QFutureWatcherBase_ProgressMaximum(const QFutureWatcherBase* self) {
	return self->progressMaximum();
}

libqt_string QFutureWatcherBase_ProgressText(const QFutureWatcherBase* self) {
	QString _ret = self->progressText();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QFutureWatcherBase_IsStarted(const QFutureWatcherBase* self) {
	return self->isStarted();
}

bool QFutureWatcherBase_IsFinished(const QFutureWatcherBase* self) {
	return self->isFinished();
}

bool QFutureWatcherBase_IsRunning(const QFutureWatcherBase* self) {
	return self->isRunning();
}

bool QFutureWatcherBase_IsCanceled(const QFutureWatcherBase* self) {
	return self->isCanceled();
}

bool QFutureWatcherBase_IsPaused(const QFutureWatcherBase* self) {
	return self->isPaused();
}

bool QFutureWatcherBase_IsSuspending(const QFutureWatcherBase* self) {
	return self->isSuspending();
}

bool QFutureWatcherBase_IsSuspended(const QFutureWatcherBase* self) {
	return self->isSuspended();
}

void QFutureWatcherBase_WaitForFinished(QFutureWatcherBase* self) {
	self->waitForFinished();
}

void QFutureWatcherBase_SetPendingResultsLimit(QFutureWatcherBase* self, int limit) {
	self->setPendingResultsLimit(limit);
}

bool QFutureWatcherBase_Event(QFutureWatcherBase* self, QEvent* event) {
	return self->event(event);
}

void QFutureWatcherBase_Started(QFutureWatcherBase* self) {
	self->started();
}

void QFutureWatcherBase_Connect_Started(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::started, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_Finished(QFutureWatcherBase* self) {
	self->finished();
}

void QFutureWatcherBase_Connect_Finished(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::finished, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_Canceled(QFutureWatcherBase* self) {
	self->canceled();
}

void QFutureWatcherBase_Connect_Canceled(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::canceled, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_Paused(QFutureWatcherBase* self) {
	self->paused();
}

void QFutureWatcherBase_Connect_Paused(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::paused, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_Suspending(QFutureWatcherBase* self) {
	self->suspending();
}

void QFutureWatcherBase_Connect_Suspending(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::suspending, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_Suspended(QFutureWatcherBase* self) {
	self->suspended();
}

void QFutureWatcherBase_Connect_Suspended(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::suspended, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_Resumed(QFutureWatcherBase* self) {
	self->resumed();
}

void QFutureWatcherBase_Connect_Resumed(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*) = reinterpret_cast<void (*)(QFutureWatcherBase*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::resumed, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QFutureWatcherBase_ResultReadyAt(QFutureWatcherBase* self, int resultIndex) {
	self->resultReadyAt(resultIndex);
}

void QFutureWatcherBase_Connect_ResultReadyAt(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*, int) = reinterpret_cast<void (*)(QFutureWatcherBase*, int)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::resultReadyAt, [self, slotFunc](int resultIndex) {
	slotFunc(self, resultIndex);
    });
}

void QFutureWatcherBase_ResultsReadyAt(QFutureWatcherBase* self, int beginIndex, int endIndex) {
	self->resultsReadyAt(beginIndex, endIndex);
}

void QFutureWatcherBase_Connect_ResultsReadyAt(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*, int, int) = reinterpret_cast<void (*)(QFutureWatcherBase*, int, int)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::resultsReadyAt, [self, slotFunc](int beginIndex, int endIndex) {
	slotFunc(self, beginIndex, endIndex);
    });
}

void QFutureWatcherBase_ProgressRangeChanged(QFutureWatcherBase* self, int minimum, int maximum) {
	self->progressRangeChanged(minimum, maximum);
}

void QFutureWatcherBase_Connect_ProgressRangeChanged(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*, int, int) = reinterpret_cast<void (*)(QFutureWatcherBase*, int, int)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::progressRangeChanged, [self, slotFunc](int minimum, int maximum) {
	slotFunc(self, minimum, maximum);
    });
}

void QFutureWatcherBase_ProgressValueChanged(QFutureWatcherBase* self, int progressValue) {
	self->progressValueChanged(progressValue);
}

void QFutureWatcherBase_Connect_ProgressValueChanged(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*, int) = reinterpret_cast<void (*)(QFutureWatcherBase*, int)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::progressValueChanged, [self, slotFunc](int progressValue) {
	slotFunc(self, progressValue);
    });
}

void QFutureWatcherBase_ProgressTextChanged(QFutureWatcherBase* self, const libqt_string progressText) {
	self->progressTextChanged(QString::fromUtf8(progressText.data, progressText.len));
}

void QFutureWatcherBase_Connect_ProgressTextChanged(QFutureWatcherBase* self, intptr_t slot) {
    void (*slotFunc)(QFutureWatcherBase*, const char*) = reinterpret_cast<void (*)(QFutureWatcherBase*, const char*)>(slot);
    QFutureWatcherBase::connect(self, &QFutureWatcherBase::progressTextChanged, [self, slotFunc](const QString& progressText) {
	slotFunc(self, progressText);
    });
}

void QFutureWatcherBase_Cancel(QFutureWatcherBase* self) {
	self->cancel();
}

void QFutureWatcherBase_SetSuspended(QFutureWatcherBase* self, bool suspendVal) {
	self->setSuspended(suspendVal);
}

void QFutureWatcherBase_Suspend(QFutureWatcherBase* self) {
	self->suspend();
}

void QFutureWatcherBase_Resume(QFutureWatcherBase* self) {
	self->resume();
}

void QFutureWatcherBase_ToggleSuspended(QFutureWatcherBase* self) {
	self->toggleSuspended();
}

void QFutureWatcherBase_SetPaused(QFutureWatcherBase* self, bool paused) {
	self->setPaused(paused);
}

void QFutureWatcherBase_Pause(QFutureWatcherBase* self) {
	self->pause();
}

void QFutureWatcherBase_TogglePaused(QFutureWatcherBase* self) {
	self->togglePaused();
}

libqt_string QFutureWatcherBase_Tr2(const char* s, const char* c) {
	QString _ret = QFutureWatcherBase::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QFutureWatcherBase_Tr3(const char* s, const char* c, int n) {
	QString _ret = QFutureWatcherBase::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QFutureWatcherBase_Delete(QFutureWatcherBase* self) {
    delete self;
}

